src/steamship/base/client.py

```diff
@@ -371,14 +371,25 @@ def call(  # noqa: C901
 
         if isinstance(response_data, dict):
             if "status" in response_data:
-                task = Task.parse_obj({**response_data["status"], "client": self})
-                # if task_resp is not None and task_resp.taskId is not None:
-                #     task = Task(client=self)
-                #     task.update(task_resp)
-                if "state" in response_data["status"]:
-                    if response_data["status"]["state"] == "failed":
-                        error = SteamshipError.from_dict(response_data["status"])
-                        logging.error(f"Client received error from server: {error}")
+                try:
+                    task = Task.parse_obj({**response_data["status"], "client": self})
+                    # if task_resp is not None and task_resp.taskId is not None:
+                    #     task = Task(client=self)
+                    #     task.update(task_resp)
+                    if "state" in response_data["status"]:
```

Comment:
```suggestion
                    if "state" in response_data["status"] and response_data["status"]["state"] == "failed"
```

The right-hand statement is only executed when the left-hand statement is True.

Reply:
Fixed this with a slight refactor instead so that we're relying on model objects instead of peeking into dicts